p2p/protocol/perf/perf.go

```diff
@@ -0,0 +1,128 @@
+package perf
+
+import (
+	"context"
+	"encoding/binary"
+	"fmt"
+	"io"
+
+	logging "github.com/ipfs/go-log/v2"
+	pool "github.com/libp2p/go-buffer-pool"
+	"github.com/libp2p/go-libp2p/core/host"
+	"github.com/libp2p/go-libp2p/core/network"
+	"github.com/libp2p/go-libp2p/core/peer"
+)
+
+var log = logging.Logger("perf")
+
+const (
+	BlockSize = 64 << 10
+
+	ID = "/perf/1.0.0"
+)
+
+type PerfService struct {
+	Host host.Host
+}
+
+func NewPerfService(h host.Host) *PerfService {
+	ps := &PerfService{h}
+	h.SetStreamHandler(ID, ps.PerfHandler)
+	return ps
+}
+
+func (p *PerfService) PerfHandler(s network.Stream) {
+	buf := pool.Get(BlockSize)
+	defer pool.Put(buf)
+
+	u64Buf := make([]byte, 8)
+	_, err := io.ReadFull(s, u64Buf)
+	if err != nil {
+		log.Errorw("err", err)
+		s.Reset()
+		return
+	}
+
+	bytesToSend := binary.BigEndian.Uint64(u64Buf)
+
+	_, err = p.drainStream(context.Background(), s, buf)
```

Comment:
This doesn't seem very useful, see my comment on the spec: https://github.com/libp2p/specs/pull/478/files#r1122634411

Reply:
Misunderstanding on my part. This is fine.